typescript/sdk/src/middleware/account/InterchainAccount.ts

```diff
@@ -56,4 +75,161 @@ export class InterchainAccount extends RouterApp<InterchainAccountFactories> {
     );
     return new InterchainAccount(helper.contractsMap, helper.multiProvider);
   }
+
+  async deployAccount(
+    chain: ChainName,
+    config: AccountConfig,
+  ): Promise<Address> {
+    const originDomain = this.multiProvider.tryGetDomainId(config.origin);
+    if (!originDomain) {
+      throw new Error(
+        `Origin chain (${config.origin}) metadata needed for deploying ICAs ...`,
+      );
+    }
+    const localRouter = this.router(this.contractsMap[chain]);
+    await this.multiProvider.handleTx(
+      chain,
+      localRouter[
+        'getDeployedInterchainAccount(uint32,address,address,address)'
+      ](
+        originDomain,
+        config.owner,
+        bytes32ToAddress(await localRouter.routers(originDomain)),
+        bytes32ToAddress(await localRouter.isms(originDomain)),
+      ),
+    );
+    const account = await localRouter[
+      'getLocalInterchainAccount(uint32,address,address,address)'
+    ](
+      originDomain,
+      config.owner,
+      bytes32ToAddress(await localRouter.routers(originDomain)),
+      bytes32ToAddress(await localRouter.isms(originDomain)),
+    );
+    if (
+      (await this.multiProvider.getProvider(chain).getCode(account)) === '0x'
+    ) {
+      throw new Error('Interchain account deployment failed');
+    }
+    return account;
+  }
+
+  // meant for ICA governance to return the populatedTx
+  async getCallRemote(
+    chain: ChainName,
+    destination: ChainName,
+    innerCalls: CallData[],
+  ): Promise<CallData> {
+    const localRouter = this.router(this.contractsMap[chain]);
+    const quote = await localRouter.quoteGasPayment(
+      this.multiProvider.getDomainId(destination),
+    );
+    const icaCall: CallData = {
+      to: localRouter.address,
+      data: localRouter.interface.encodeFunctionData(
+        'callRemote(uint32,(bytes32,uint256,bytes)[])',
+        [
+          this.multiProvider.getDomainId(destination),
+          innerCalls.map((call) => ({
+            to: addressToBytes32(call.to),
+            value: call.value,
+            data: call.data,
+          })),
+        ],
+      ),
+      value: quote,
+    };
+    return icaCall;
+  }
+
+  async getAccountOwner(
+    chain: ChainName,
+    account: Address,
+  ): Promise<[number, AddressBytes32]> {
+    return this.router(this.contractsMap[chain]).accountOwners(account);
+  }
+
+  // general helper for different overloaded callRemote functions
+  async callRemote(
```

Comment:
any way you can dedupe this with `getCallRemote`?

Reply:
difficult because getCallRemote and handleTx takes in ContractTransaction

Reply:
maybe there's a way to contract a ContractTransaction from PopulatedTransaction but not sure if it's worth spending more time on it

Reply:
`ContractTransaction` is after the tx has been submitted to the network
this is a lot of duplicated code
the only difference is instead of `handleTx` you should use `multiProvider.sendTransaction`

Reply:
fixed